src/Orchard/Tasks/Locking/Services/DistributedLockService.cs

```diff
@@ -141,9 +141,19 @@ public class DistributedLockService : Component, IDistributedLockService {
 
             ExecuteOnSeparateTransaction(repository => {
                 // Try to find a valid lock record in the database.
-                var record = repository.Table.FirstOrDefault(x => x.Name == internalName && (x.ValidUntilUtc == null || x.ValidUntilUtc >= _clock.UtcNow));
+                var records = repository.Table.Where(x => x.Name == internalName).ToList();
+                var record = records.FirstOrDefault(x => x.ValidUntilUtc == null || x.ValidUntilUtc >= _clock.UtcNow);
                 if (record == null) {
-                    // No record existed, so we're good to create a new one.
+                    if (records.Any()) {
```

Comment:
@Skrypt I insist. :) That's not why it's redundant. It's redundant because the `records` collection is empty, the loop will simply skip. The `foreach` does the empty checking, so no real reason to do it twice.